src/drive/components/IntentButton.jsx

```diff
@@ -0,0 +1,68 @@
+import styles from '../styles/intentbutton'
+
+/* global cozy */
+import React from 'react'
+import classNames from 'classnames'
+
+import Modal from 'cozy-ui/react/Modal'
+
+class IntentButton extends React.Component {
+  state = {
+    modalIsOpen: false
+  }
+
+  openModal() {
+    this.setState({
+      modalIsOpen: true
+    })
+  }
+
+  closeModal = () => {
+    this.setState({
+      modalIsOpen: false
+    })
+  }
+
+  render() {
+    const { className, data, intent, children } = this.props
+    const { modalIsOpen } = this.state
+    return (
+      <span className={classNames(styles['intentButton'])}>
+        {modalIsOpen && (
+          <Modal secondaryAction={() => this.closeModal()}>
+            <Intent data={data} intent={intent} closeModal={this.closeModal} />
+          </Modal>
+        )}
+        <button
+          className={className}
+          onClick={() => this.openModal(data, intent)}
```

Comment:
I think you forgot to remove `data, intent` from the function call, as it doesn't appear to be needed :)